core-tests/shared/src/test/scala/zio/FiberSpec.scala

```diff
@@ -162,7 +162,38 @@ object FiberSpec extends ZIOBaseSpec {
             currentFiber <- ZIO.succeed(Fiber.currentFiber())
           } yield assertTrue(currentFiber.isDefined)
         }
-      } @@ TestAspect.fromLayer(Runtime.enableCurrentFiber)
+      } @@ TestAspect.fromLayer(Runtime.enableCurrentFiber),
+      suite("FiberFailure stack trace handling")(
+        test("FiberFailure captures the full stack trace including user code") {
+          def subcall(): Unit =
+            Unsafe.unsafe { implicit unsafe =>
+              Runtime.default.unsafe.run(ZIO.fail("boom")).getOrThrowFiberFailure()
```

Comment:
The coverage here is not great; it's only testing against `ZIO.fail(???: String)`. We also need tests both for: `ZIO.fail`, `ZIO.die`, `Exit.fail`, `Exit.die` and `ZIO.interrupt`.

Note that for `fail`, we need to test against both having a `String` and a `Throwable` as a failure